### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by Mimetis** [Mimetis/Dotmim.Sync#527] [General]
`v0.8.0` is in progress dev.
You can grab the pre-release here : https://www.nuget.org/packages?q=Dotmim.Sync

## Upgrade to version v0.8.0

Do not forget to call the `UpgradeAsync `method to upgrade your current schema version the last version.

- **Client Side** : No need to call `UpgradeAsync`, since it's handled by default by `DMS`
- **Server Side** : You need to create a `RemoteOrchestrator` with the correct `SyncSetup` and call the `UpgradeAsync` method

``` cs
// Upgrade to last version
if (await remoteOrchestrator.NeedsToUpgradeAsync())
   await remoteOrchestrator.UpgradeAsync(progress:progress);
```

When making the upgrade of your system (server / clients), be sure to remember that:
- A client upgraded **is not compatible** with a non upgraded server.
- A server upgraded **will be compatible** with non upgraded clients

That being said, be sure to firstly upgrade your server to the last version, call the `UpgradeAsync` (as explained before), then upgrade your clients.

Once upgraded your schema should be updated to the last version

## New features and issues fixed

This version is focused on web performances.
- **IMPROVEMENT**: The `Dotmim.Sync.Web.Client` and `Dotmim.Sync.Web.Server` have been re writed to be more robust and efficient. (https://github.com/Mimetis/Dotmim.Sync/issues/523)
![image](https://user-images.githubusercontent.com/4592555/113772787-73f5ba00-9725-11eb-9a8a-9e282eb0257c.png)

- **NEW** : When using a `WebClientOrchestrator` orchestrator, you have now a new argument `maxDownladingDegreeOfParallelism` (default to `4`) that allows your client application to make http requests in parrallel.
- **IMPROVEMENT**: Custom serializer are now used everywhere, even when generating a snapshot.
- **REGRESSION** : A server can't handle multi serializers anymore. Once configured on both side, only one unique serializer can be used.
   - This regression is due to the fact that the snapshot serialization should be available for anyone.
- **NEW** : Serializers options are now directly available from `SyncOptions` on both side (server / client)
- **BACKWARD COMPATIBILITY** : a client < `v0.8.0` will continue to work with a server upgraded to `v0.8.0`
   - If you have a snapshot available, until all your clients are migrated to `v0.8.0`, the snapshot MUST be generated with a json serializer

**Comment by maxrodro** [comment]
Hi @Mimetis 

Excellent work is done to this tool, I am surprised how quickly you get a version with several improvements that help us all.

I thank you for your effort to this valuable project that you have. :-)

**Reply by Mimetis** [reply]
You should test it before applauding, I guess :D
A lot of changes here, internally, so be sure to test the beta version and wait the release to perform your upgrade :)

(And thanks a lot your kind words, appreciate ! )

**Comment by tommy-uytterhaegen** [comment]
It just keeps getting more awesome every day :) Looking forward to the production release of 0.8.0 and thanks for keeping the client backwards compatible!

**Reply by Mimetis** [reply]
That's pretty hard :D

**Reply by tommy-uytterhaegen** [reply]
I can imagine, but I really appreciate it. I dread the day you need to break the backwards compatibility because - in my case - the user chooses when he upgrades and a lot of them need some time to install a new version :/

**Reply by Mimetis** [reply]
The deal is to always have a backward compatibility with the last version, from the client to the server side :)
Not simple :)
